geps/gep-1619/index.md

```diff
@@ -457,58 +583,44 @@ considered cautiously, as making associations to Gateway API's notion of Gateway
-multiple routes to satisfy different session persistence requirements for each individual route path.
+Session persistence configuration specified in a route rule SHALL override equivalent configuration in `BackendLBPolicy`.
+In this situation, implementations MAY want to indicate a warning via a log or status. Refer to [GEP-713](/geps/gep-713)
+and/or [GEP-2648](/geps/gep-2648) for more specific details on how to handle override scenarios.
 
-Edge cases will arise when providing supporting services, routes, and the route rule section. For guidance on addressing
-conflicting attachments, please consult the [Expected API Behavior](#expected-api-behavior) section, which outlines API
-use cases.
-
-To learn more about the process of attaching to services, routes, and the route rule section, please refer to
-[GEP-713](/geps/gep-713).
+Edge cases will arise when implementing session persistence support for both backends and route rules through
+`BackendLBPolicy` and the route rule's `sessionPersistence` field. For guidance on addressing conflicting
+attachments, please consult the [Edge Case Behavior](#edge-case-behavior) section, which outlines API
+use cases. Only a subset of implementations have already designed their data plane to incorporate route-level session
+persistence, making it likely that route-level session persistence will be less widely implemented.
 
 ### Traffic Splitting
 
-In scenarios involving traffic splitting, session persistence impacts load balancing done after routing, regardless if
-it is attached to a service or a route. When a persistent session is established and traffic splitting is configured
-across services, the persistence to a single backend should be maintained across services. Consequently, a persistent
-session takes precedence over traffic split weights when selecting a backend after route matching. It's important to
-note that session persistence does not impact the process of route matching.
+In scenarios involving traffic splitting, session persistence impacts load balancing done after routing. When a
+persistent session is established and traffic splitting is configured across services, the persistence to a single
+backend MUST be maintained across services. Consequently, a persistent session takes precedence over traffic split
+weights when selecting a backend after route matching. It's important to note that session persistence does not impact
+the process of route matching.
 
 When using multiple backends in traffic splitting, all backend services should have session persistence enabled.
-Nonetheless, implementations should also support traffic splitting scenarios in which one service has persistence
+Nonetheless, implementations MUST also support traffic splitting scenarios in which one service has persistence
 enabled while the other does not. This support is necessary, particularly in scenarios where users are transitioning
 to or from an implementation version designed with or without persistence.
```

Comment:
Idealy, we should have all backend services enable session persistence in traffic splitting scenarios. However, I want to get some clarity on this traffic splitting scenario of one service (svc1) has persistence enabled while the other (svc2, svc3) does not.  In this scenario, if we keep session persistence only applied to svc1, all traffic will eventually goes to svc1 because the persistence should be maintained, resulting in traffic overloading svc1 and may break the system.  A more reasonable interpretation of this is that session persistence MUST be applied to all backends (svc1, svc2, svc3) involved in traffic splitting if any of them have session persistence enabled.

Reply:
I think you have a good point that it may be much more practical to apply session persistence to all backends given one backend has session persistence configured, but like @costinm mentioned, I think it might be too prescriptive to say it must be done that way.

Do you see any problem with leaving the spec open for implementations to do both?:
1. Maintain only session persistence to svc1 (eventually transitioning all traffic to svc1, possibly overloading it)
2. Apply session persistence to all svc1-3

Reply:
Let's leave the spec open and have implementations to decide which behavior they want since it's an edge case.

Reply:
Done in a new commit: https://github.com/kubernetes-sigs/gateway-api/pull/2634/commits/13eaa83fb171a836cf59496b57bbff31145351a3